### PR TITLE
r2modman: update electron version

### PR DESCRIPTION
**Pull request by aidalgol**
## Description of changes
We had been pinning to an older Electron release to avoid regressions in later Electron versions, but that version is now EOL.

## Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by huantianad — APPROVED** [review]

**Comment by huantianad** [comment]
Has the convention for mainProgram been changed to always set it?

**Comment by aidalgol** [comment]
> Has the convention for mainProgram been changed to always set it?

I'm not sure on that one.  The [nixpkgs manual](https://nixos.org/manual/nixpkgs/unstable/#var-meta-mainProgram) doesn't specify either way.  I'm fine going either way: if it's redundant, we can safely drop it, but it's also almost nil maintenance overhead to set it AFAICT.

**Review by Gerg-L — APPROVED** [review]

**Review by bddvlpr — APPROVED** [review]

**Comment by bddvlpr** [comment]
No isssues so far, seems to work with electron 26+ now.

**Comment by Gerg-L** [comment]
> Has the convention for mainProgram been changed to always set it?

Yes due to `lib.getExe` no longer making assumptions based on `pname`

**Comment by Gerg-L** [comment]
> but it's also almost nil maintenance overhead to set it AFAICT.

Unless the name of the program changes, which is not common

**Comment by huantianad** [comment]
> > Has the convention for mainProgram been changed to always set it?
> 
> Yes due to `lib.getExe` no longer making assumptions based on `pname`

Makes sense, previously I was told to not set it if it was equal to pname, but with getExe that makes sense

**Comment by drupol** [comment]
Setting `meta.mainProgram` from `pname` is wrong, it shouldn't be a variable but a string literal.

**Comment by wegank** [comment]
> Setting `meta.mainProgram` from `pname` is wrong, it shouldn't be a variable but a string literal.

Is there a reference in nixpkgs for this best practice? Personally, I have no argument for suggesting this when the derivation is written using `stdenv.mkDerivation rec`, since `meta.mainProgram` is then not overridable.

**Comment by drupol** [comment]
For sure, using `pname` as a shortcut is nice for DRY but creates a binding that goes too far.

I've switched to `finalAttrs` pattern in here: https://github.com/NixOS/nixpkgs/pull/275496 feel free to give a review

File: pkgs/games/r2modman/default.nix [hunk 1]
```diff
@@ -114,5 +114,6 @@ stdenv.mkDerivation rec {
     license = licenses.mit;
     maintainers = with maintainers; [ aidalgol huantian ];
     inherit (electron.meta) platforms;
+    mainProgram = pname;
```
Comment (SuperSandro2000) [inline pkgs/games/r2modman/default.nix:117]: The point of mainProgram was not to inherit pname which could have a suffix but to set a literal string.

Comment (drupol) [reply pkgs/games/r2modman/default.nix:117]: Fixed in https://github.com/NixOS/nixpkgs/pull/275499

**Pull request merged**